From f3daa5bcb6bcb105cbea8997d7d8943d86148970 Mon Sep 17 00:00:00 2001
From: Sanyog Kale <skale@habana.ai>
Date: Fri, 01 Apr 2022 14:08:58 +0300
Subject: [PATCH] [SW-77251] gaudi3: add function to get crc count

In order to verify image or perform sha verification, crc count
needs to be subtracted from total image. This patch adds function
to get crc count which will be used by erom functions as well.

Change-Id: I3e57831f0252f2c891c22430563f9b194908aec7
---

diff --git a/arch/gaudi3/arc/common.c b/arch/gaudi3/arc/common.c
index 66fe3c1..ae970cf 100644
--- a/arch/gaudi3/arc/common.c
+++ b/arch/gaudi3/arc/common.c
@@ -443,6 +443,17 @@
 	regmap_writel(cold_rst_data.data, mmCOLD_RST_FLOPS_2);
 }
 
+uint32_t arcboot_rom_get_num_crc(uint32_t size)
+{
+	uint32_t num_crc;
+
+	num_crc = size / MAX_DMA_SIZE_CRC;
+	if (size % MAX_DMA_SIZE_CRC)
+		num_crc++;
+
+	return num_crc;
+}
+
 #define PPBOOT_MAX_IMAGE_SIZE	(2 * MAX_DMA_SIZE_CRC)
 static int arcboot_rom_bringup_img_sub(int img, struct boot_desc *desc, enum fw_post_reset_cmd upd_status,
 		uint8_t cur_img)
@@ -499,9 +510,7 @@
 	if (!security_efuse_image_validation_en())
 		return ARCBOOT_BRINGUP_OK;
 
-	num_crc = desc->img_size / MAX_DMA_SIZE_CRC;
-	if (desc->img_size % MAX_DMA_SIZE_CRC)
-		num_crc++;
+	num_crc = arcboot_rom_get_num_crc(desc->img_size);
 
 	act_img_size = desc->img_size;
 	act_img_size = act_img_size - (num_crc * CRC32_SIZE);
diff --git a/arch/gaudi3/arc/dev_common.h b/arch/gaudi3/arc/dev_common.h
index f02e43d..80ebb87 100644
--- a/arch/gaudi3/arc/dev_common.h
+++ b/arch/gaudi3/arc/dev_common.h
@@ -8,3 +8,4 @@
 #define GPIOS_PER_BANK			16
 
 void arcboot_rom_notify_slave(void);
+uint32_t arcboot_rom_get_num_crc(uint32_t size);
